From fe72ec5883bf21286ca8e046f7a5c30cb0ff48b4 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:45 -0400
Subject: [virt] kvm/nept: Advertise WB type EPTP

Message-id: <1406040016-3289-71-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86149
O-Subject: [RHEL7 PATCH v2 070/221] KVM: nEPT: Advertise WB type EPTP
Bugzilla: 1048496 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

At least WB must be possible.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Reviewed-by: Gleb Natapov <gleb@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit ca72d970ff9aaf6f7e321ead9d4775c5aa0cdb0b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index f5fce57..287fdc5 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2251,8 +2251,8 @@ static __init void nested_vmx_setup_ctls_msrs(void)
  if (enable_ept) {
   /* nested EPT: emulate EPT also to L1 */
   nested_vmx_secondary_ctls_high |= SECONDARY_EXEC_ENABLE_EPT;
-  nested_vmx_ept_caps = VMX_EPT_PAGE_WALK_4_BIT;
-  nested_vmx_ept_caps |= VMX_EPT_INVEPT_BIT;
+  nested_vmx_ept_caps = VMX_EPT_PAGE_WALK_4_BIT |
+    VMX_EPTP_WB_BIT | VMX_EPT_INVEPT_BIT;
   nested_vmx_ept_caps &= vmx_capability.ept;
   /*
    * Since invept is completely emulated we support both global
-- 
1.7.1